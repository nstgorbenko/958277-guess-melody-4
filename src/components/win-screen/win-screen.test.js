import WinScreen from "./win-screen.jsx";

import React from "react";
import renderer from "react-test-renderer";

describe(`WinScreen Component rendering`, () => {
  describe(`3 questions`, () => {
    it(`renders with 0 mistakes`, () => {
      const tree = renderer
        .create(
            <WinScreen
              questionsCount={3}
              mistakesCount={0}
              onReplayButtonClick={() => {}}
            />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`renders with 1 mistake`, () => {
      const tree = renderer
        .create(
            <WinScreen
              questionsCount={3}
              mistakesCount={1}
              onReplayButtonClick={() => {}}
            />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`2 questions`, () => {
    it(`renders with 0 mistakes`, () => {
      const tree = renderer
        .create(
            <WinScreen
              questionsCount={2}
              mistakesCount={0}
              onReplayButtonClick={() => {}}
            />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it(`renders with 1 mistake`, () => {
      const tree = renderer
        .create(
            <WinScreen
              questionsCount={2}
              mistakesCount={1}
              onReplayButtonClick={() => {}}
            />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
