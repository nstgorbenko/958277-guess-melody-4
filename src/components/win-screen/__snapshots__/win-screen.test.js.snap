// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinScreen Component rendering 2 questions renders with 0 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    2 вопроса
     и совершили 
    0 ошибок
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;

exports[`WinScreen Component rendering 2 questions renders with 1 mistake 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    1 вопрос
     и совершили 
    1 ошибку
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;

exports[`WinScreen Component rendering 3 questions renders with 0 mistakes 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    3 вопроса
     и совершили 
    0 ошибок
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;

exports[`WinScreen Component rendering 3 questions renders with 1 mistake 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    2 вопроса
     и совершили 
    1 ошибку
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;
